File: 1935_Maximum_Number_of_Words_You_Can_Type.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        broken_set = set(brokenLetters)
        words = text.split()
        count = 0
        
        for word in words:
            word_set = set(word)
            if word_set.isdisjoint(broken_set):                   #if not (word_set & broken):
                count += 1
                
        return count


a = Solution()
print(a.canBeTypedWords("hello world", "ad"))  # Output: 1
print(a.canBeTypedWords("leet code", "lt"))  # Output: 1
print(a.canBeTypedWords("leet code", "e"))  # Output: 0

File: 1935_Maximum_Number_of_Words_You_Can_Type.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int canBeTypedWords(char* text, char* brokenLetters) {
    int count = 0;
    char* word;
    char* rest = text;
    int broken[26] = {0};

    // Mark broken letters
    for (int i = 0; brokenLetters[i] != '\0'; i++) {
        broken[brokenLetters[i] - 'a'] = 1;
    }

    // Split text into words and check each word
    while ((word = strtok_r(rest, " ", &rest))) {
        int canType = 1;
        for (int i = 0; word[i] != '\0'; i++) {
            if (broken[word[i] - 'a']) {
                canType = 0;
                break;
            }
        }
        if (canType) {
            count++;
        }
    }

    return count;
    
}

int main() {
    char text[] = "hello world";
    char brokenLetters[] = "ad";
    int result = canBeTypedWords(text, brokenLetters);
    printf("Result: %d\n", result); // Expected output: 1    

    return 0;
}